lib/ts/recipe/emailverification/components/features/emailVerification/index.tsx

```diff
@@ -51,15 +51,15 @@ export const EmailVerification: React.FC<Prop> = (props) => {
     const fetchIsEmailVerified = useCallback(async () => {
         const token = getQueryParams("token") ?? undefined;
         if (token === undefined) {
-            if (!sessionContext.doesSessionExist) {
+            if (sessionContext.loading === false && !sessionContext.doesSessionExist) {
                 await props.recipe.config.redirectToSignIn(props.history);
             } else {
                 // we check if the email is already verified, and if it is, then we redirect the user
                 return (await props.recipe.recipeImpl.isEmailVerified({ userContext })).isVerified;
             }
```

Comment:
Shouldn't we do this entire thing only if `sessionContext.loading === false`? As opposed to doing `return (await props.recipe.recipeImpl.isEmailVerified({ userContext })).isVerified;` if `sessionContext.loading === true`

Reply:
You are correct, but this will only ever run if `loading` is false.  Let's discuss this along with `useOnMountAPICall` on a call tomorrow.

Reply:
I'll add an assert instead above and a comment explaining.